android/quest/src/main/java/org/smartregister/fhircore/quest/data/patient/PatientRepository.kt

```diff
@@ -243,16 +258,31 @@ constructor(
   }
 
   suspend fun getQuestionnaire(questionnaireResponse: QuestionnaireResponse): Questionnaire {
-    return if (questionnaireResponse.questionnaire != null) {
-      val questionnaireId = questionnaireResponse.questionnaire.split("/")[1]
-      loadQuestionnaire(questionnaireId = questionnaireId)
-    } else {
-      Timber.e(
-        Exception(
-          "Cannot open QuestionnaireResponse because QuestionnaireResponse.questionnaire is null"
+    return when {
+      questionnaireResponse.questionnaire.isNullOrBlank() -> {
+        Timber.e(
+          Exception(
+            "Cannot open QuestionnaireResponse because QuestionnaireResponse.questionnaire is null"
+          )
         )
-      )
-      Questionnaire()
+        Questionnaire()
+      }
+      else -> {
+        val questionnaireUrlList = questionnaireResponse.questionnaire.split("/")
+        when {
+          questionnaireUrlList.isNotEmpty() && questionnaireUrlList.size > 1 -> {
+            loadQuestionnaire(questionnaireId = questionnaireUrlList[1])
+          }
+          else -> {
```

Comment:
remove all this.. its highly unlikely that once validated for non empty questionnaire it would not have any url.. just pick last list item and it would be questionnaire url

Reply:
I removed `questionnaireUrlList.isNotEmpty()` Check and keep the other one `questionnaireUrlList.size > 1` to make sure it doesn't throw IndexOutOfBoxException in case of invalid URL.